repos/system_upgrade/el7toel8/actors/verifycheckresults/actor.py

```diff
@@ -13,24 +13,26 @@ class VerifyCheckResults(Actor):
     """
 
     name = 'verify_check_results'
-    consumes = (CheckResult, Inhibitor)
+    consumes = (Report,)
     produces = ()
     tags = (ReportPhaseTag, IPUWorkflowTag)
 
     def process(self):
-        results = list(self.consume(CheckResult, Inhibitor))
-        errors = [msg for msg in results if msg.severity == 'Error']
+        results = list(self.consume(Report))
+        inhibitors = [msg for msg in results if 'inhibitor' in msg.flags]
+        high_sev_msgs = [msg for msg in results if msg.severity == 'high' and 'inhibitor' not in msg.flags]
+        msgs_to_report = inhibitors+high_sev_msgs
 
         report_file = '/tmp/leapp-report.txt'
-        error = report.generate_report(results, report_file)
+        error = report.generate_report(msgs_to_report, report_file)
         if error:
             self.report_error('Report Error: ' + error)
         else:
             self.log.info('Generated report at ' + report_file)
 
-        if errors:
-            for e in errors:
-                self.report_error('%s: %s: %s' % (e.severity, e.result, e.summary))
+        if inhibitors:
+            for e in inhibitors:
+                self.report_error('%s: %s: %s' % (e.title, e.severity, e.detail['summary']))
```

Comment:
I might be wrong, but I think `e.detail['summary']` is not working as you'd expect it. If you didn't decode the json string.

Reply:
Did not tested it yet but you are probably right. Shame we do not have flag between WIP and untested :)

Reply:
@Rezney untested by author = WIP to me

Reply:
Fair enough. Basically I just wanted to move with this a little while the framework PR is still being evaluated so we can identify possible caveats...